src/main/java/io/jenkins/plugins/tuleap_api/steps/TuleapSendTTMResultsStep.java

```diff
@@ -0,0 +1,133 @@
+        return credentialId;
+    }
+
+    @Override
+    public StepExecution start(StepContext stepContext) throws Exception {
+        return new Execution(stepContext, this);
+    }
+
+    public static class Execution extends SynchronousStepExecution<Void> {
+        private final static long serialVersionUID = 1L;
+        private final transient TuleapSendTTMResultsStep tuleapSendTTMResultsStep;
+        private final transient TuleapConfiguration tuleapConfiguration;
+        private final transient TuleapSendTTMResultsRunner tuleapSendTTMResultsRunner;
+        private final transient Run run;
+        private final transient PrintStream logger;
+        private final  TaskListener taskListener;
+        private final  FilePath filePath;
+        private final  EnvVars envVars;
+
+        protected Execution(StepContext context, TuleapSendTTMResultsStep tuleapSendTTMResultsStep) throws IOException, InterruptedException {
+            super(context);
+            this.tuleapSendTTMResultsStep = tuleapSendTTMResultsStep;
+            this.taskListener = context.get(TaskListener.class);
+            this.logger = taskListener.getLogger();
+            this.filePath = getContext().get(FilePath.class);
+            this.envVars = getContext().get(EnvVars.class);
+            this.run = getContext().get(Run.class);
+
+            final Injector injector = Guice.createInjector(new TuleapApiGuiceModule());
+            this.tuleapSendTTMResultsRunner = injector.getInstance(TuleapSendTTMResultsRunner.class);
+            this.tuleapConfiguration = TuleapConfiguration.get();
+        }
+
+        @Override
+        protected Void run() throws Exception {
+            assert filePath != null;
+
+            logger.println("Retrieving Tuleap API credentials");
+            final TuleapAccessToken tuleapAccessToken = CredentialsProvider.findCredentialById(
+                tuleapSendTTMResultsStep.getCredentialId(),
+                TuleapAccessToken.class,
+                run,
+                URIRequirementBuilder.fromUri(tuleapConfiguration.getApiBaseUrl()).build()
+            );
+
+            assert tuleapAccessToken != null;
+
+            tuleapSendTTMResultsRunner.run(
+                tuleapAccessToken,
+                logger,
+                tuleapSendTTMResultsStep,
+                filePath,
+                envVars
+            );
+
+            return null;
+        }
+    }
+
+    @Extension
+    public static class DescriptorImpl extends StepDescriptor {
+
+        @Override
+        public Set<? extends Class<?>> getRequiredContext() {
+            return new HashSet<>(
+                Arrays.asList(
+                    TaskListener.class,
+                    FilePath.class,
+                    EnvVars.class,
+                    Run.class
+                )
+            );
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "TuleapSendTTMResults";
```

Comment:
It should start with lower case letter. 
It seems ( but not sure ) that "broke"  the snippet generator when we want to generate the command